lib/ukboot/early_init.c

```diff
@@ -0,0 +1,150 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/argparse.h>
+#include <uk/assert.h>
+#include <uk/boot/earlytab.h>
+#include <uk/essentials.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/prio.h>
+
+#if CONFIG_LIBUKLIBPARAM
+#include <uk/libparam.h>
+#endif /* CONFIG_LIBUKLIBPARAM */
+
+char *boot_argv[CONFIG_LIBUKBOOT_MAXNBARGS];
+int boot_argc;
+
+/* Copy of bi->cmdline, as the boot protocol buffer may
+ * be unmapped by paged memory init.
+ */
+static char cmdline[CONFIG_LIBUKBOOT_CMDLINE_MAX_LEN];
+static __sz cmdline_len;
+
+/* Mutable version of the cmdline. This is tokenized by uk_argnparse().
+ * Keep this as a global as `argv` points to tokens in this string.
+ */
+static char cmdline_argv[CONFIG_LIBUKBOOT_CMDLINE_MAX_LEN];
```

Comment:
Could we find a different name than `cmdline_argv`. With `_argv` I am biased to read `char *argv[]` 🤔

Reply:
Plus, see my questions below.

Reply:
Will rename to `cmdline_argparse`